tensorrt_llm/_torch/custom_ops/cute_dsl_custom_ops.py

```
@@ -2789,3 +2789,229 @@ def _(
+        """
+        # Validate SM version
+        sm_version = get_sm_version()
+        if sm_version < 100:
+            raise ValueError(
+                f"CuTE DSL top-k requires Blackwell (SM100+), but got SM {sm_version}. "
+            )
+
+        # Validate inputs
+        if top_k > 2048:
+            raise ValueError(
+                f"CuTE DSL top-k supports max top_k=2048, but got {top_k}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)